.travis.yml

```diff
@@ -12,7 +12,7 @@ addons:
 matrix:
   include:
     - os: linux
-      env: CUDA=8.0.61-1 CMAKE_OPTIONS="-DPENGUINV_BUILD_TEST=ON"
+      env: CUDA=8.0.61-1 CMAKE_OPTIONS="-DPENGUINV_BUILD_TEST=ON -DPENGUINV_ENABLE_PNG_SUPPORT=ON -DPENGUINV_USE_EXTERNAL_PNG=OFF"
```

Comment:
I thought we just need to create another file and move the logic there and then include this file in CMake file for unit tests?

Reply:
I moved the logic in extra cmake files or maybe I'm misunderstanding you. Anyway I have issue on Appveyor I still need to fix.

Reply:
What I meant is: we move installation of libpng for Win32 and Unix in a separate file. If we are unable to install it we set a flag (macro), let's call it `PENGUINV_DISABLE_PNG`. This is C++ macro, not environment variable.

Reply:
Something like this:
Create a separate CMake file with initial logic which was in Cmake for unit tests. Put it, for example, in `script/cmake_lib` folder. In unit tests Cmake add a line of code `include("${CMAKE_CURRENT_LIST_DIR}/../../script/cmake_lib/png.cmake")`.

Reply:
> Something like this:
> Create a separate CMake file with initial logic which was in Cmake for unit tests. Put it, for example, in `script/cmake_lib` folder. In unit tests Cmake add a line of code `include("${CMAKE_CURRENT_LIST_DIR}/../../script/cmake_lib/png.cmake")`.

I created two CMake modules `cmake/external_zlib.cmake` and `cmake/external_png.cmake` and I'm calling
```
include(external_zlib)
include(external_png)
```
in `test/unit_tests/CMakeLists.txt`. For it's exactly what you're requiring. I can rename the files/folders later on if the builds are successful.

Reply:
Do it even better: why don't you include `external_zlib` inside `external_png` in unit tests cmake we include only one as png requires zlib? :)

Reply:
> The idea behind is that we enable `png` support by default but disable it when we cannot install the library :) It's an opposite idea what you have right now.
> It's better to do this way because as an user won't want to tune some variables and etc.

PNG support is enabled by default. We are not installing anything on UNIX platforms. On Windows, as long as CMake is able to download the library, everything is fine. I just don't know how to indicate a CMake download failure with a macro. Feel free to edit the pr, if you know how to do it. Currently, I'm not even sure if moving the CMake scripts to separate files will work on Windows.

Reply:
> Do it even better: why don't you include `external_zlib` inside `external_png` in unit tests cmake we include only one as png requires zlib? :)

Will merge as soon as Appveyor builds properly.